yt/data_objects/construction_data_containers.py

```diff
@@ -950,6 +950,28 @@ def _fill_particles(self, part):
         for p in part:
             self[p] = self._data_source[p]
 
+    def _check_sph_type(self, fi):
+        field = fi.name
+        sph_ptypes = self.ds._sph_ptypes
+        raise_error = False
+        ptype = sph_ptypes[0]
+        if field[0] in self.ds.known_filters:
+            if field[0] not in sph_ptypes:
+                raise_error = True
+            else:
+                ptype = field[0]
+        elif fi.is_alias:
+            if fi.alias_name[0] not in sph_ptypes:
+                raise_error = True
+            elif field[0] != "gas":
+                ptype = field[0]
+        else:
+            if fi.name[0] not in sph_ptypes and fi.name[0] != "gas":
+                raise_error = True
+        if raise_error:
+            raise KeyError(f"{field[0]} is not a SPH particle type!")
```

Comment:
It might be easier to follow if we instead instantiated the error early as 
```python
err = KeyError(f"{field[0]} is not a SPH particle type!")
```

and replaced every `raise_error = True` with `raise err`

Reply:
Done